spyderlib/plugins/__init__.py

```diff
@@ -264,6 +270,7 @@ def __init__(self, main=None, **kwds):
         """Bind widget to a QMainWindow instance"""
         super(SpyderPluginMixin, self).__init__(**kwds)
         assert self.CONF_SECTION is not None
+        self.PLUGIN_PATH = Path(inspect.getfile(self.__class__)).parent

```

Comment:
I made all changes (plus a small update to how io plugins load) except this part...